packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -674,6 +685,23 @@ void ArbCore::operator()() {
                 machine = getMachineUsingStateKeys<MachineThread>(
                     *tx, pending_checkpoint.machine_state_keys, cache);
             } else {
+                // Cache pre-sideload machines
+                if (last_assertion.sideloadBlockNumber) {
+                    auto block = *last_assertion.sideloadBlockNumber;
+                    std::lock_guard<std::mutex> lock(sideload_cache_mutex);
+                    sideload_cache[block] = std::make_unique<Machine>(*machine);
+                    auto it = sideload_cache.begin();
+                    while (it != sideload_cache.end()) {
```

Comment:
I think it'd be good to add a comment here explaining this logic (doesn't need to be too detailed)